sympy/physics/control/control_plots.py

```diff
@@ -863,18 +865,35 @@ def bode_phase_numerical_data(system, initial_exp=-5, final_exp=5, freq_unit='ra
     x, y = LineOver1DRangeSeries(phase,
         (_w, 10**initial_exp, 10**final_exp), xscale='log', **kwargs).get_points()
 
+    if phase_unwrap:
+        if(phase_unit == 'rad'):
+            for i in range(1, len(y)):
+                diff = y[i] - y[i - 1]
+                if diff > pi:      # Jump from -pi to pi
+                    y[i] = (y[i] - 2*pi)
+                elif diff < -pi:   # Jump from pi to -pi
+                    y[i] = (y[i] + 2*pi)
+
+        elif(phase_unit == 'deg'):
+            for i in range(1, len(y)):
+                diff = y[i] - y[i - 1]
+                if diff > 180:      # Jump from -180 to 180
+                    y[i] = (y[i] - 360)
+                elif diff < -180:   # Jump from 180 to -180
+                    y[i] = (y[i] + 360)
```

Comment:
Any reason for not using [Numpy's unwrap](https://numpy.org/doc/stable/reference/generated/numpy.unwrap.html)?

Reply:
Yup, we are aiming to remove all numerical methods from numpy in `control_plots.py`. I consciously did not use `numpy.unwrap()`.

Reply:
The plotting module uses Numpy, both when `adaptive=True` and `adaptive=False`. Therefore, this line of code is using Numpy:

```py
x, y = LineOver1DRangeSeries(phase,
        (_w, 10**initial_exp, 10**final_exp), xscale='log', **kwargs).get_points()
```

If `adaptive=True`, `x, y` are lists of Python's `float`. If `adaptive=False`, `x, y` are Numpy arrays with `dtype=float`. Which means there is no problem in using Numpy's `unwrap`.

> we are aiming to remove all numerical methods from numpy in control_plots.py. 

Imho, this statement needs to be carefully weighted on a case by case basis. I can understand that logic for something like `pole_zero_plot`, where [it was proven the superiority](https://github.com/sympy/sympy/issues/25234#issuecomment-1593319448) of SymPy's `roots`  vs Numpy's `roots`. But if you go down the "no-numpy" route, you'd have to:

1. remove `LineOver1DRangeSeries`
2. compute symbolic values from symbolic expressions, probably using `subs`
3. then converting them to float for plotting, thus losing any "accuracy" advantage of the symbolic values.

Ending up with a terribly slow control plotting module, probably barely usable even with today's hardware. What would be the advantages of this approach?

@moorepants @ Sc0rpi0n101  What are your opinions?

Reply:
I don't think we are aiming to "remove all numpy", as sympy plotting relies on numpy and matplotlib, but any plot methods in the control package should ideally be built on sympy's plotting tools. If we aren't working with numpy and matplotlib directly in the control package that is the best, but it may mean new things need to be implemented in sympy.plot.

Reply:
> The plotting module uses Numpy, both when `adaptive=True` and `adaptive=False`. Therefore, this line of code is using Numpy:

I have pointed something similar out in our earlier discussions that functions like `LineOver1DRangeSeries` already use `Numpy` so it is not possible to totally eliminate the dependency. In some cases like roots, `Sympy` gives better precision than `Numpy` so we replace numerical methods with algebraic methods and sacrifice on time.

> > we are aiming to remove all numerical methods from numpy in control_plots.py.
> 
> Imho, this statement needs to be carefully weighted on a case by case basis. 

Yes totally agreed.